core/dbt/main.py

```diff
@@ -1006,7 +1006,8 @@ def parse_args(args, cls=DBTArgumentParser):
         """,
     )
 
-    p.add_argument(
+    warn_error_flag = p.add_mutually_exclusive_group()
```

Comment:
Is this the argparse way to say `--warn-error` + `--warn-error-options` are mutually exclusive? I'm guessing there's also a `click`-y way to do it?

I'm thinking we'll need to add a check within `flags.py` as well, because it's possible that one of these configs was supplied as a CLI flag, and the other was set as an env var or UserConfig attr